src/lib/server/database.ts

```diff
@@ -54,7 +55,33 @@ export const collections = {
 	bucket,
 };
 
+/**
```

Comment:
Below is my proposed way to run database migrations. So far, there was no need for migrations. Since this PR introduces new field `assistant.searchTokens`, we need all existing assistants to have this new field (i.e. run a migration script)

Reply:
But how will this work for people running chat-ui locally? Ideally they shouldn't be running migrations script manually to get the latest changes? 

I guess in that case it's fairly minor (you just loose search on assistants) but is there a better way to run these automatically if needed?

Reply:
> But how will this work for people running chat-ui locally?

they would need to run the script manually once by declaring env vars `migrations=refreshSearchTokens` before `npm run [xyz]`.

> is there a better way to run these automatically if needed?

I might be wrong but it might be bit complicated to set up an automaitc mechanism. You'd need some kind of trigger to run it automatically, in which case it is as bad as running the script manually. OR we can setup some mechanism based on git (check if `database-migrations.ts` file was changed & then run the changed migration function automoatically; however, it seems overkill to me)

Reply:
I took a look at how [jellyfin](https://github.com/jellyfin/jellyfin) handles migration (a big open source project that users self-host and that needs to trigger migrations on user setups from time to time). Looks like this is how they handle migrations.

1. Each migration is represented by some kind of object that contains a GUID and a function that performs the migration. (see [example](https://github.com/jellyfin/jellyfin/blob/master/Jellyfin.Server/Migrations/Routines/FixPlaylistOwner.cs))
2. A list of migrations is stored in the user DB (list of GUIDs), starts empty.
4. A routine starts when the server starts, checks if there are any migrations that have not run yet (by comparing the GUIDs stored in the DBs vs the list of migrations in the code) [(see code)](https://github.com/jellyfin/jellyfin/blob/0bc41c015f4ec907de75fe215589b7e30a819b54/Jellyfin.Server/Migrations/MigrationRunner.cs#L108-L143)
6. The routine runs migrations that have not yet run, and adds their GUID in the DB when it's done, so they can be skipped the next time.

I know this is more complex and definitely unrelated to the search feature, but I think having an automated migrations setup would be very useful and I don't think we can ask users to run migrations scripts manually, this is different from moon-landing in that sense 😅 

I can do the work on this if we decide to go for it, don't want to add more to your plate for something unrelated to the PR

Reply:
Maybe we can finish reviewing this PR as is, merging it with the custom migrations script so it's not blocking and I can implement the automated migrations later if we want to.

Is there an easy way to fall back to the old search if the migration hasn't run yet? (as a temporary fix for chat-ui users)

Reply:
yep, followup PR specifically for migrations sounds great

Reply:
That would probably involve DB locks as well (to prevent running the same migration twice simultaneously)